tapeagents/llm_function.py

```diff
@@ -145,12 +151,14 @@ def generate_steps(self, agent, tape: Tape, llm_stream: LLMStream):
         # TODO: streaming
         # TODO: more robust parsing that doesn't rely on ':' and '\n'
         output_text = llm_stream.get_text()
-        prompt_text = llm_stream.prompt.messages[0]["content"]
-        text = prompt_text + "\n" + output_text
-        output_lines = text.split("\n")[-len(self.outputs) :]
+        output_lines = output_text.split("\n")[-len(self.outputs) :]
         output_values = [output_lines[0]] + [line.split(":")[1].strip() for line in output_lines[1:]]
-        for output, value in zip(self.outputs, output_values):
-            yield output.parse(value)
+        for i, output in enumerate(self.outputs):
```

Comment:
We'll need to talk about this code fragment.

Reply:
parsing free-form llm_output for different model gets REALLY tricky... I created a more explicit data extraction and I added unit tests. Let me know what you think!